src/messages/MOSDPing.h

```diff
@@ -100,21 +101,24 @@ class MOSDPing : public Message {
     ::encode(peer_stat, payload);
     ::encode(stamp, payload);
 
-    bufferlist pad;
-    size_t s = MAX(min_message_size - payload.length(), 0);
-    // this should be big enough for normal min_message padding sizes. since
-    // we are targetting jumbo ethernet frames around 9000 bytes, 16k should
-    // be more than sufficient!  the compiler will statically zero this so
-    // that at runtime we are only adding a bufferptr reference to it.
-    static char zeros[16384] = {};
-    while (s > sizeof(zeros)) {
-      pad.append(buffer::create_static(sizeof(zeros), zeros));
-      s -= sizeof(zeros);
-    }
+    size_t s = 0;
+    if (min_message_size > payload.length())
+      s = min_message_size - payload.length();
+    ::encode((uint32_t)s, payload);
     if (s) {
-      pad.append(buffer::create_static(s, zeros));
+      // this should be big enough for normal min_message padding sizes. since
+      // we are targetting jumbo ethernet frames around 9000 bytes, 16k should
+      // be more than sufficient!  the compiler will statically zero this so
+      // that at runtime we are only adding a bufferptr reference to it.
+      static char zeros[16384] = {};
+      while (s > sizeof(zeros)) {
+	payload.append(buffer::create_static(sizeof(zeros), zeros));
+	s -= sizeof(zeros);
+      }
+      if (s) {
```

Comment:
It's not really redundant. First "if" (line 108) checks whether there's extra payload to be added, "while" loop (#114) decreases value of "s" by sizeof(zeros), leaving possibility for "s" to be nonzero when exiting a loop and second "if" (line 118) append remaining - less than or equal to sizeof(zeros) -  payload.